keyboards/hackpad/config.h

```diff
@@ -0,0 +1,42 @@
+// Copyright 2023 Nico Stuhlmueller (@ThePurox)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+/*
+ * Feature disable options
+ *  These options are also useful to firmware size reduction.
+ */
+
+/* disable debug print */
+//#define NO_DEBUG
+
+/* disable print */
+//#define NO_PRINT
+
+/* SK6812 RGB LED */
+#define RGB_DI_PIN D4
+
+#define ENCODERS_PAD_A {F4}
+#define ENCODERS_PAD_B {F5}
+#define ENCODER_DIRECTION_FLIP
+
+#ifdef RGBLIGHT_ENABLE
+#define RGBLED_NUM 16 // Number of LEDs
+#define DRIVER_LED_TOTAL RGBLED_NUM
+#define RGBLIGHT_SLEEP /* If defined, the RGB lighting will be switched off when the host goes to sleep */
+#define RGBLIGHT_EFFECT_BREATHING
+#define RGBLIGHT_EFFECT_RAINBOW_MOOD
+#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
+#define RGBLIGHT_EFFECT_SNAKE
+#define RGBLIGHT_EFFECT_KNIGHT
+#define RGBLIGHT_EFFECT_CHRISTMAS
+#define RGBLIGHT_EFFECT_STATIC_GRADIENT
+#define RGBLIGHT_EFFECT_RGB_TEST
+#define RGBLIGHT_EFFECT_ALTERNATING
+#define RGBLIGHT_EFFECT_TWINKLE
+#define RGBLIGHT_LIMIT_VAL 255
+#define RGB_VAL_STEP 12
+#define RGBLIGHT_DEFAULT_SPD 144
+#define RGBLIGHT_DEFAULT_MODE RGBLIGHT_MODE_RAINBOW_SWIRL + 2
+#endif
```

Comment:
Most (if not all) of this can be moved to json

Reply:
Now it is